server/sonar-main/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -69,30 +69,53 @@ public void close() {
     allProcessesCommands.close();
   }
 
-  @Override
-  public ProcessMonitor launch(EsCommand esCommand) {
-    Process process = null;
-    try {
-      cleanupOutdatedEsData(esCommand);
-      writeConfFiles(esCommand);
-      ProcessBuilder processBuilder = create(esCommand);
-      logLaunchedCommand(esCommand, processBuilder);
+  public ProcessMonitor launch(AbstractCommand command) {
 
-      process = processBuilder.start();
+    ElasticsearchConfiguration fileSystem = command.getElasticsearchConfiguration();
+    if (fileSystem != null) {
+      cleanupOutdatedEsData(fileSystem);
+      writeConfFiles(fileSystem);
+    }
+
+    // FIXME refactor this
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)